R/position-dodge.R

```diff
@@ -117,8 +117,10 @@ PositionDodge <- ggproto("PositionDodge", Position,
     if (identical(self$preserve, "total")) {
       n <- NULL
     } else {
-      n <- vec_unique(data[c("group", "PANEL", "xmin")])
-      n <- vec_group_id(n[c("PANEL", "xmin")])
+      data$xmin <- data$xmin %||% data$x
+      cols <- intersect(colnames(data), c("group", "PANEL", "xmin"))
```

Comment:
Since "group" and "PANEL" are required data columns and you just defined the xmin column, is there any instances where this would yield anything different than `c("group", "PANEL", "xmin")`?

Reply:
I guess if both "xmin" and "x" are initially `NULL` - is that what is the intend?

Reply:
The column-name intersection is just to protect against any potentially missing columns. I imagine a determined extension could avoid creating `group` and `PANEL` columns.

Reply:
If so, it would break in a multitude of other places as both columns are assumed to exist all over the place

Reply:
If we assume all well-behaved extensions, then still xmin/x can be `NULL` as you pointed out. But then there'd be multiple downstream assumptions that no longer hold too.

Reply:
It is not necessary to assume well-behaved extensions. The extensions would never work in any settings if they removed these columns so they _cannot_ exist in any meaningful way. If there is a chance that min/x is NULL then this should be guarded by a proper error

Reply:
Alright I've thrown in a `check_required_aesthetics()` in here